pkg/controller/subscription/subscription_controller.go

```diff
@@ -253,24 +245,36 @@ func (r *ReconcileSubscription) HandleUpgrades(ctx context.Context, rhmiSubscrip
 
 	isServiceAffecting := rhmiConfigs.IsUpgradeServiceAffecting(latestRHMICSV)
 
-	// checks if there is changes in the rhmiconfig cr and recalculate the status
-	if reschedule, ok := config.Annotations[v1alpha1.RecalculateScheduleAnnotation]; ok && reschedule == "true" {
-		err = rhmiConfigs.UpdateStatus(ctx, r.client, config, latestRHMIInstallPlan, rhmiSubscription.Status.CurrentCSV)
-		if err != nil {
-			return reconcile.Result{}, err
+	if isServiceAffecting {
+
+		newUpgradeAvailable := &integreatlyv1alpha1.UpgradeAvailable{
+			TargetVersion: rhmiSubscription.Status.CurrentCSV,
+			AvailableAt:   latestRHMIInstallPlan.CreationTimestamp,
 		}
 
-		config.Annotations[v1alpha1.RecalculateScheduleAnnotation] = ""
-		err = r.client.Update(ctx, config)
-		if err != nil {
+		if !reflect.DeepEqual(newUpgradeAvailable, config.Status.UpgradeAvailable) {
+
+			config.Status.UpgradeAvailable = newUpgradeAvailable
+			if err := r.client.Status().Update(context.TODO(), config); err != nil {
+				return reconcile.Result{}, err
+			}
+			return reconcile.Result{
+				Requeue:      true,
+				RequeueAfter: 10 * time.Second,
```

Comment:
```suggestion
				RequeueAfter: 1 * time.Second,
```

1 second should be enough for the RHMIConfig controller to reconcile the status